api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -638,12 +639,17 @@ func (c *imageBuilder) createKanikoJobSpec(
 	activeDeadlineSeconds := int64(c.config.BuildTimeoutDuration / time.Second)
 	var volumes []v1.Volume
 	var volumeMounts []v1.VolumeMount
-	var envVar []v1.EnvVar
+	var envVars []v1.EnvVar
 
 	// Configure additional credentials for specific image registries and artifact services
 	kanikoArgs = c.configureKanikoArgsToAddCredentials(kanikoArgs)
 	volumes, volumeMounts = c.configureVolumesAndVolumeMountsToAddCredentials(volumes, volumeMounts)
-	envVar = c.configureEnvVarsToAddCredentials(envVar)
+	envVars = c.configureEnvVarsToAddCredentials(envVars)
```

Comment:
nit: I assume this is to get the S3 credential, but cmiiw, based on the previous #605 MR, the args is expected to be passed in the KanikoAdditionalArgs, so perhaps can modify [this chunk of note](https://github.com/caraml-dev/merlin/blob/main/api/config/config.go#L461-L462), if we're going to change it this way?

Reply:
Ahh, yes the motivation was to allow the image building job to get access to the S3 credentials. Technically it's still possible to pass the credentials via `KanikoAdditionalArgs` but the only downside is that the secret might be exposed if `KanikoAdditionalArgs` is specified explicitly within a certain repo (like what we do). I'll reword the comments you mentioned a little to suggest multiple ways to pass the S3 credentials (including the one introduced in this PR). Thanks for suggesting this!